src/Database/LSMTree/Internal/MergeSchedule.hs

```diff
@@ -162,19 +165,34 @@ data Level s h = Level {
   , residentRuns :: !(V.Vector (Run s h))
   }
 
--- TODO: proper instance
-deriving via OnlyCheckWhnfNamed "Level" (Level s h) instance NoThunks (Level s h)
+instance ( forall a. NoThunks a => NoThunks (MutVar s a)
+         ) => NoThunks (Level s (Handle h)) where
+  showTypeOf (_ :: Proxy (Level s (Handle h))) = "Level"
+  wNoThunks ctx
+    (Level (a :: MergingRun s (Handle h)) (b :: V.Vector (Run s (Handle h))))
+    = allNoThunks [noThunks ctx a, noThunks ctx b]
 
 -- | A merging run is either a single run, or some ongoing merge.
 data MergingRun s h =
     MergingRun !(MergingRunState s h)
   | SingleRun !(Run s h)
 
+instance ( forall a. NoThunks a => NoThunks (MutVar s a)
```

Comment:
arguably, `nothunks` should provide instances for types in `primitive`; `nothunks` already depends / provides intances for `vector`.

Reply:
True, I've created https://github.com/input-output-hk/nothunks/issues/57 and https://github.com/input-output-hk/nothunks/issues/56 for the missing instances